panther_manager/include/panther_manager/safety_manager_node.hpp

```diff
@@ -40,76 +38,75 @@ using BatteryStateMsg = sensor_msgs::msg::BatteryState;
 using BoolMsg = std_msgs::msg::Bool;
 using DriverStateMsg = panther_msgs::msg::DriverState;
 using IOStateMsg = panther_msgs::msg::IOState;
-using LEDAnimationMsg = panther_msgs::msg::LEDAnimation;
 using SystemStatusMsg = panther_msgs::msg::SystemStatus;
 
-class ManagerBTNode : public rclcpp::Node
+/**
+ * @brief This class is responsible for creating a BehaviorTrees responsible for safety and shutdown
+ * management, spining them, and updating blackboard entries based on subscribed topics.
+ */
+class SafetyManagerNode : public rclcpp::Node
 {
 public:
-  ManagerBTNode(
+  SafetyManagerNode(
     const std::string & node_name, const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
-  ~ManagerBTNode() {}
+  ~SafetyManagerNode() {}
 
   void Initialize();
 
 protected:
   void DeclareParameters();
   void RegisterBehaviorTree();
-  void CreateLightsTree();
   void CreateSafetyTree();
   void CreateShutdownTree();
+
+  /**
+   * @brief Checks whether the required blackboard entries for the lights tree are present. These
+   * entries are usually updated when the first ROS message containing required information is
+   * received.
+   *
+   * @return True if all required blackboard entries are present.
+   */
   bool SystemReady();
 
+  BT::Tree safety_tree_;
+  BT::NodeStatus safety_tree_status_ = BT::NodeStatus::IDLE;
+  BT::NodeStatus shutdown_tree_status_ = BT::NodeStatus::IDLE;
+
 private:
   void BatteryCB(const BatteryStateMsg::SharedPtr battery);
   void DriverStateCB(const DriverStateMsg::SharedPtr driver_state);
   void EStopCB(const BoolMsg::SharedPtr e_stop);
   void IOStateCB(const IOStateMsg::SharedPtr io_state);
   void SystemStatusCB(const SystemStatusMsg::SharedPtr system_status);
   void SafetyTreeTimerCB();
-  void LightsTreeTimerCB();
 
   void ShutdownRobot(const std::string & reason);
 
   static constexpr float kCriticalBatteryTemp = 55.0;
   static constexpr float kFatalBatteryTemp = 62.0;
 
-  bool launch_shutdown_tree_;
   float update_charging_anim_step_;
-  std::optional<unsigned> battery_status_;
-  std::optional<unsigned> battery_health_;
-  std::optional<bool> e_stop_state_;
-  std::optional<IOStateMsg::SharedPtr> io_state_;
 
   rclcpp::Subscription<BatteryStateMsg>::SharedPtr battery_sub_;
   rclcpp::Subscription<DriverStateMsg>::SharedPtr driver_state_sub_;
   rclcpp::Subscription<BoolMsg>::SharedPtr e_stop_sub_;
   rclcpp::Subscription<IOStateMsg>::SharedPtr io_state_sub_;
   rclcpp::Subscription<SystemStatusMsg>::SharedPtr system_status_sub_;
-  rclcpp::TimerBase::SharedPtr lights_tree_timer_;
   rclcpp::TimerBase::SharedPtr safety_tree_timer_;
 
   BT::BehaviorTreeFactory factory_;
-  BT::NodeConfig lights_config_;
   BT::NodeConfig safety_config_;
   BT::NodeConfig shutdown_config_;
-  BT::NodeStatus lights_tree_status_;
-  BT::NodeStatus safety_tree_status_;
-  BT::NodeStatus shutdown_tree_status_;
-  BT::Tree lights_tree_;
-  BT::Tree safety_tree_;
   BT::Tree shutdown_tree_;
```

Comment:
It was not necessary to expose this tree in private class members. I guess I can move it in case we want to create test for shutdown tree in the future